### PR TITLE
appbookstore_C001_step6

### PR DESCRIPTION
**Pull request by Poikka001**

### CODE REVIEW

File: src/main/java/mate/academy/intro/controller/AuthController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/auth")
+public class AuthController {
+    private final UserService userService;
+
+    @PostMapping("/registration")
+    public UserResponseDto registrationUser(@RequestBody UserRegistrationRequestDto requestDto)
```
Comment (Rommelua) [inline src/main/java/mate/academy/intro/controller/AuthController.java:20]: Don't forget to use data validation in the new DTO classes

File: src/main/java/mate/academy/intro/dto/user/UserRegistrationRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package mate.academy.intro.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+import org.hibernate.validator.constraints.Length;
+
+@Data
+public class UserRegistrationRequestDto {
```
Comment (Rommelua) [inline src/main/java/mate/academy/intro/dto/user/UserRegistrationRequestDto.java:9]: Add FieldMatch annotation for checking that password and repeatPassword field in the UserRegistrationRequestDto are the same

File: src/main/java/mate/academy/intro/mapper/BookMapper.java [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
     Book toDto(CreateRequestBookDto requestDto);
 }
+//{
+//  "email": "john.doe@example.com",
+//  "password": "securePassword123",
+//  "repeatPassword": "securePassword123",
+//  "firstName": "John",
+//  "lastName": "Doe",
+//  "shippingAddress": "123 Main St, City, Country"
+//}
```
Comment (Rommelua) [inline src/main/java/mate/academy/intro/mapper/BookMapper.java:21]: delete this

File: src/main/java/mate/academy/intro/mapper/UserMapper.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import org.mapstruct.Mapping;
+
+@Mapper(config = MapperConfig.class)
+public interface UserMapper {
+
+    @Mapping(target = "id", source = "id")
+    @Mapping(target = "email", source = "email")
+    @Mapping(target = "firstName", source = "firstName")
+    @Mapping(target = "lastName", source = "lastName")
+    @Mapping(target = "shippingAddress", source = "shippingAddress")
```
Comment (Rommelua) [inline src/main/java/mate/academy/intro/mapper/UserMapper.java:16]:
```suggestion
```

File: src/main/java/mate/academy/intro/exception/RegistrationException.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package mate.academy.intro.exception;
+
+public class RegistrationException extends Exception{
```
Comment (sokimaaa) [inline src/main/java/mate/academy/intro/exception/RegistrationException.java:3]: use unchecked exception here

File: src/main/java/mate/academy/intro/controller/AuthController.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/auth")
+public class AuthController {
+    private final UserService userService;
+
+    @PostMapping("/registration")
```
Comment (sokimaaa) [inline src/main/java/mate/academy/intro/controller/AuthController.java:20]: "/auth/registration"? Pretty weird endpoint

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 7]
```diff
@@ -1,3 +1,5 @@
 databaseChangeLog:
   - include:
       file: db/changelog/changes/01-create-books-table.yaml
+  - include:
```
Comment (sokimaaa) [inline src/main/resources/db/changelog/db.changelog-master.yaml:4]: update liquibase changelog with 03-inster-users script

**Review by kerrrusha — APPROVED** [review]

**Pull request merged**